src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -175,13 +184,31 @@ public bool TryGenerateBashScript(BuildScriptGeneratorContext context, out strin
             return resultPlatforms;
         }
 
-        private static string GetBenvArgs(Dictionary<string, string> benvArgsMap)
+        private static string GetBenvArgs(IDictionary<string, string> benvArgsMap)
         {
             var listOfBenvArgs = benvArgsMap.Select(t => $"{t.Key}={t.Value}");
             var benvArgs = string.Join(' ', listOfBenvArgs);
             return benvArgs;
         }
 
+        private void RunCheckers(
+            BuildScriptGeneratorContext ctx,
+            IDictionary<string, string> tools,
+            [NotNull] List<ICheckerMessage> checkerMessageSink)
+        {
+            var checkers = _checkers.WhereApplicable(tools);
+
+            using (var timedEvent = _logger.LogTimedEvent("RunCheckers"))
+            {
+                timedEvent.AddProperty(
+                    "checkersApplied",
+                    string.Join(',', checkers.Select(checker => checker.GetType().Name)));
+
+                checkerMessageSink.AddRange(checkers.SelectMany(checker => checker.CheckSourceRepo(ctx.SourceRepo)));
```

Comment:
I would place the calls to `CheckSourceRepo` and `CheckToolVersions` in a try-catch block...we do not want these to fail the build even though the actual build might have succeeded